### PR TITLE
Keydoc to keypair

### PR DESCRIPTION
**Pull request by osinakayah**

### CODE REVIEW

File: packages/core/lib/services/credential/service.js [hunk 1]
```diff
@@ -79,11 +81,17 @@ class CredentialService {
...
-    const kp = getKeypairFromDoc(keyDoc);
-    kp.signer = kp.signer();
-    const suite = getSuiteFromKeyDoc(kp);
     vp.setHolder(keyDoc.controller);
-    return vp.sign(suite, challenge, domain);
+    keyDoc.keypair = keyDocToKeypair(keyDoc, getDock());
+    return vp.sign(
+      {
+        ...keyDoc,
+        id: `${keyDoc.controller}#keys-1`, // HACK: make it work with SDK did resolution, this is an SDK limitation
```
Comment (SamHellawell) [inline packages/core/lib/services/credential/service.js:89]: this wont work for all keydocs, you cant assume the id in wallet's case. you have to use keydoc.id which should previously set the correct value for the DID key

File: packages/core/lib/services/credential/service.js [hunk 2]
```diff
@@ -79,11 +81,17 @@ class CredentialService {
     for (const signedVC of credentials) {
       vp.addCredential(signedVC);
     }
-    const kp = getKeypairFromDoc(keyDoc);
-    kp.signer = kp.signer();
-    const suite = getSuiteFromKeyDoc(kp);
     vp.setHolder(keyDoc.controller);
-    return vp.sign(suite, challenge, domain);
+    keyDoc.keypair = keyDocToKeypair(keyDoc, getDock());
+    return vp.sign(
```
Comment (SamHellawell) [inline packages/core/lib/services/credential/service.js:86]: why was this change required when vp.sign using suite has been working just fine afaik?

Comment (osinakayah) [reply packages/core/lib/services/credential/service.js:86]: it does not work if the key type is Sr25519

**Review by SamHellawell — APPROVED** [review]
